api_app/api/routes/resource_helpers.py

```diff
@@ -40,26 +55,68 @@ async def save_and_deploy_resource(resource: Resource, resource_repo: ResourceRe
             user=user,
             resource_template=resource_template,
             resource_template_repo=resource_template_repo,
-            action=RequestAction.Install)
+            action=RequestAction.Install,
+        )
         return operation
     except Exception as e:
         resource_repo.delete_item(resource.id)
         logging.error(f"Failed send resource request message: {e}")
-        raise HTTPException(status_code=status.HTTP_503_SERVICE_UNAVAILABLE, detail=strings.SERVICE_BUS_GENERAL_ERROR_MESSAGE)
+        raise HTTPException(
+            status_code=status.HTTP_503_SERVICE_UNAVAILABLE,
+            detail=strings.SERVICE_BUS_GENERAL_ERROR_MESSAGE,
+        )
 
 
-def mask_sensitive_properties(properties: Dict[str, Any], template: ResourceTemplate) -> dict:
+def mask_sensitive_properties(
+    properties: Dict[str, Any], template: ResourceTemplate
+) -> dict:
     updated_resource_parameters = properties
 
-    for prop_name, prop in template.properties.items():
-        if prop.sensitive is True and prop_name in properties:
-            updated_resource_parameters = {**properties}
-            updated_resource_parameters[prop_name] = strings.REDACTED_SENSITIVE_VALUE
+    flattened_template_props = {}
+
+    # recusrse the template and flatten all properties into a list
+    def flatten_template_props(template_fragment: dict):
+        if template_fragment is None:
+            return
+
+        for prop_name, prop in template_fragment.items():
+            if (
+                prop_name == "properties"
+                and template_fragment["properties"] is not None
+            ):
+                for inner_prop_name, inner_prop in template_fragment[
+                    "properties"
+                ].items():
+                    flattened_template_props[inner_prop_name] = inner_prop
+
+            if isinstance(prop, list) and len(prop) > 0:
+                for p in prop:
+                    if isinstance(p, dict):
+                        flatten_template_props(p)
+
+            if isinstance(prop, dict) and prop_name != "if":
+                flatten_template_props(prop)
+
+    flatten_template_props(template.dict())
+
+    def recurse_input_props(prop_dict: dict):
```

Comment:
nit: maybe name it `recurse_mask_sensitive_props`? Or something else that hints at sensitive properties being redacted in this method

Reply:
it doesn't just recurse sensitive props though, it recurses _all_ props, and just redacts the sensitive ones

Reply:
(it's also an inner method, inside the `mask_sensitive_props` method